### PR TITLE
Add a new property "lightbox" to enable/disable lightbox feature.

### PR DESCRIPTION
**Pull request by momocow**
Hi,

I've made a new property `lightbox` to enable/disable the feature, i.e., user interaction with images. Also, I've modified the demo app to include a checkbox for the new property. But I'm not really sure what tests I should take into consideration.

Sorry I found no contribution guide from README; therefore I file the PR directly. Feel free to close this if you are not planning to accept any PRs.

### CODE REVIEW

File: src/lib/components/Grid.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ import PropTypes from "prop-types";
 import Lightbox from "./Lightbox";
 import "./Grid.css";
 
-const Grid = ({ images, rowHeight, margin = 0, width }) => {
+const Grid = ({ images, rowHeight, margin = 0, width, lightbox = true }) => {
```
Comment (AustinGomez) [inline src/lib/components/Grid.js:6]: Would prefer `isLightboxEnabled` for this :)

**Review by AustinGomez — CHANGES_REQUESTED** [review]
Thanks for contributing. Looks great! Just a small nitpick. 

Also I should really work on adding a contribution guide.

File: src/lib/components/Grid.js (hunk 1 above)
Comment (momocow) [reply src/lib/components/Grid.js]: Sure :)

**Comment by momocow** [comment]
@AustinGomez I've been confused when writing tests for the new feature. Hope to get some responses from you, thank you. 😅 

Before I start to test the new property `isLightboxEnabled` can really prevent the lightbox from display, I think I should write a test to show that an user click can really trigger the lightbox to be shown.

To write tests for user interaction, I've read from SOF that I should rather trust the state than depend on the render of the triggered component, i.e., Lightbox. That is, instead of testing with `expect(wrapper.find(".lightbox-window")).toHaveLength(1);` after a click event, I should test whether `onLightboxOpen` handler is called or not, suppose that there is also a property `onLightboxOpen` on the Grid component.

That is what I think I should do before doing tests for the Grid component.

Since a Lightbox is an **internal** component living inside a Grid, I think there are some essential properties missing on the Grid component to give the full control on the component and make tests easier.  The missing properties are `onLightboxOpen`, `onLightboxNext`, `onLightboxPrev`, `onLightboxClose`.

I can pause this PR to work on the missing properties in another PR, and after the PR accepted then I can continue to finish tests for this PR.

What do you think about it? :)

**Comment by AustinGomez** [comment]
I see, so just checking you'd add the `onLightbox*` properties, add tests to verify that these are called (or not), and then add tests for the actual functions you're passing in to these when enabled?

That sounds good, very clean. You can do it in another PR if you'd like, otherwise adding it to this one is fine I'm not particularly picky.

**Comment by momocow** [comment]
> apologies for the delay on this, it's been a busy week for me

Never mind. I've got quite a lot of works to do, too.

This component is for my side project, so I make it a bit lower priority, but I'll finish the PR asap! 🤣

**Comment by AustinGomez** [comment]
closing due to inactivity

**Pull request closed without merging**